spec/lib/thread_printer_spec.rb

```diff
@@ -20,14 +20,19 @@
       # The documentation calls out that the backtrace is an array or
       # nil.
       expect(described_class).to receive(:puts).with(/--BEGIN THREAD DUMP, .*/)
-      expect(described_class).to receive(:puts).with(/Thread: #<InstanceDouble.*>/)
+      expect(described_class).to receive(:puts).with(/Thread: #<Thread:.*>/)
       expect(described_class).to receive(:puts).with(/Created at: .*/)
       expect(described_class).to receive(:puts).with(nil)
       expect(described_class).to receive(:puts).with(/--END THREAD DUMP, .*/)
-      th = instance_double(Thread, backtrace: nil)
+      q = Queue.new
+      th = Thread.new { q.pop }
+      expect(th).to receive(:backtrace).and_return(nil)
       expect(th).to receive(:[]).with(:created_at).and_return(Time.now - 30)
       expect(Thread).to receive(:list).and_return([th])
+      allow(Thread).to receive(:list).and_call_original
```

Comment:
why not expect [`.at least(:once)`, I wish there were something shorter]? it's kinda wordy but at least you can delete it if the implementation changes such that it's not needed anymore.

Reply:
I'm a bit puzzled: what about the thread dumper necessitates non-determinism in listing threads? This so the leaked thread checker can run, right?

Reply:
I can try removing the `allow` line. Maybe it was needed in an earlier version where the leaked thread check was in an `after` hook and not in the `around` hook.